src/vcpkg/paragraphs.cpp

```diff
@@ -482,11 +482,11 @@ namespace vcpkg::Paragraphs
         if (auto p = pghs.get())
         {
             BinaryControlFile bcf;
-            bcf.core_paragraph = BinaryParagraph(p->front());
+            bcf.core_paragraph = BinaryParagraph(std::move(p->front()));
             p->erase(p->begin());
 
-            bcf.features =
-                Util::fmap(*p, [&](auto&& raw_feature) -> BinaryParagraph { return BinaryParagraph(raw_feature); });
+            bcf.features = Util::fmap(
+                *p, [&](auto&& raw_feature) -> BinaryParagraph { return BinaryParagraph(std::move(raw_feature)); });
```

Comment:
I don't believe this change is correct. Note that `auto&&` is perfect forwarding, not move.

Reply:
Doesn't compile without move

Reply:
I see. How about something like:

```
        ExpectedL<std::vector<Paragraph>> maybe_paragraphs = get_paragraphs(fs, package_dir / "CONTROL");
        if (auto pparagraphs = maybe_paragraphs.get())
        {
            auto& paragraphs = *pparagraphs;
            BinaryControlFile bcf;
            bcf.core_paragraph = BinaryParagraph(std::move(paragraphs[0]));
            if (bcf.core_paragraph.spec != spec)
            {
                return msg::format(msgMismatchedSpec,
                                   msg::path = package_dir,
                                   msg::expected = spec,
                                   msg::actual = bcf.core_paragraph.spec);
            }

            bcf.features.reserve(paragraphs.size() - 1);
            for (std::size_t idx = 1; idx < paragraphs.size(); ++idx)
            {
                bcf.features.emplace_back(BinaryParagraph{std::move(paragraphs[idx])});
            }

            return bcf;
        }

        return maybe_paragraphs.error();
```
?

Reply:
My understanding is that `reserve()` specifies the absolute capacity of the vector and not `size() + new_capacity`

[cppreference](https://en.cppreference.com/w/cpp/container/vector/reserve):

> Increase the capacity of the vector (the total number of elements that the vector can hold without requiring reallocation) **to** a value that's greater or equal to new_cap.

Note that it says __to__ as opposed to __by__.

Reply:
Thanks for the suggestion!